pkgs/by-name/ou/outfly/package.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  fetchFromGitea,
+  buildType ? "release",
+  platform ? "linux",
+  enableWayland ? true,
+  enableXorg ? true,
+  enableOpenGL ? true,
+  enableVulkan ? true,
+  rustPlatform,
+  makeDesktopItem,
+  pkg-config,
+  libxkbcommon,
+  alsa-lib,
+  libGL,
+  vulkan-loader,
+  wayland,
+  libXrandr,
+  libXcursor,
+  libX11,
+  libXi,
+  ...
+}:
+
+assert (platform == "linux") -> (enableWayland || enableXorg);
+assert (platform == "linux") -> (enableVulkan || enableOpenGL);
+
+rustPlatform.buildRustPackage rec {
+  pname = "outfly";
+  inherit buildType;
+  version = "0.10.1";
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "outfly";
+    repo = "outfly";
+    rev = "v${version}";
+    hash = "sha256-VmUiDORTlXQvAKNXSQOI4+xqp/uhqBcKGHdzhM+D1pU=";
+
+  };
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [
+    "embed_assets"
+  ] ++ lib.optionals enableXorg [ "x11" ] ++ lib.optionals enableWayland [ "wayland" ];
+
+  runtimeInputs =
+    [ libxkbcommon ]
+    ++ lib.optionals enableOpenGL [ libGL ]
+    ++ lib.optionals enableXorg [
+      libXrandr
+      libX11
+    ]
+    ++ lib.optionals enableVulkan [ vulkan-loader ];
+
+  buildInputs = [
+    alsa-lib.dev
+    libXcursor
+    libXi
+  ] ++ lib.optionals enableWayland [ wayland ];
+
+  nativeBuildInputs = [ pkg-config ];
+  doCheck = false;
+
+  postFixup = ''
+    patchelf $out/bin/outfly \
+    --add-rpath ${lib.makeLibraryPath runtimeInputs}
+  '';
+
+  postPatch = ''
+    find ../cargo-vendor-dir -name Cargo.toml -exec sed -i 's/^workspace = true//' {} +
+  '';
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "outfly";
+      exec = "outfly";
+
+      desktopName = "OutFly";
+      categories = [ "Game" ];
+    })
+  ];
+  meta = {
+    description = "A breathtaking 3D space game in the rings of Jupiter";
+    homepage = "https://yunicode.itch.io/outgly";
+    downloadPage = "https://codeberg.org/outfly/outfly/releases";
+    changelog = "https://codeberg.org/outfly/outfly/releases/tag/v${version}";
+    license = lib.licenses.gpl3;
```

Comment:
the code is licensed under GPL-3 but some assets are licensed under CC-BY-4.0 and should probably be added.

Reply:
Assets are not just under CC-BY-4.0, but also:

- CC-BY-3.0
- CC-BY-SA-3.0
- [Pixabay Content License](https://pixabay.com/service/license-summary)
- SIL OPEN FONT LICENSE Version 1.1
- CC0 and public domain (not sure if this is relevant)
- [A custom license](https://bjj.mmedia.is/acknow.html), stating "[the assets] are publicly available. You do not need a special permission to use them"

Reply:
@zimward thank you for the quick review!

have been thinking about this and im not sure what the correct approach is here.

The assumption here was that the entire game and its source could be treated as GPL-3. All assets appear to be under licenses that should allow for adaptations/derivative work to be licensed in a GPL-3 compatible manner.

If the game and its source can not be be considered a single work for this purpose, or this violates the intent of the meta.license attribute im not sure what to do about the Pixabay license.
It would require a custom license and it seems to have a number of restrictions on unmodified works, including forbidding selling it, making the package at probably at least unfree.
But im happy to add licenses for the assets if that is the correct way

Unsurprisingly im not a lawyer and my knowledge of copyright and how nixpkgs approaches licenses is limited.
If you can point at any guidance on how to use licenses in nixpkgs that would be appreciated.

Reply:
generally all licenses should be added as they are. if the Pixabay license makes it `unfree` ,`unfreeRedistributeable` or `free` is not something i can answer either. i asked on the nixos matrix channel, if no one answers in the next few days maybe ping emily.

Reply:
i will keep an eye on the matrix channel, and otherwise follow up directly with the author of the repo about use and licensing of assets, thanks

Reply:
The [Pixabay licence](https://pixabay.com/service/license-summary/), for reference. The two iffy parts seem to be:

> You cannot sell or distribute Content (either in digital or physical form) on a Standalone basis. Standalone means where no creative effort has been applied to the Content and it remains in substantially the same form as it exists on our website.

IIRC there is some font licence or similar that also prohibits verbatim redistribution that is considered FOSS because you can just make some trivial modification or bundle it with something else. So that might not be an issue. On the other hand:

> You cannot use Content in any immoral or illegal way, especially Content which features recognisable people.
> You cannot use Content in a misleading or deceptive way.
> You cannot use any of the Content as part of a trade-mark, design-mark, trade-name, business name or service mark.

Some of these use restrictions seem potentially problematic (e.g. “immoral” has shades of the infamous JSON “Don’t Be Evil” clause). So I’d say that the conservative thing to do would be to include the other normal licences and then a `lib.licenses.unfree` (or `lib.licenses.unfreeRedistributable`, but usually that implies permission for verbatim redistribution) for this Pixabay one.

It’s a pretty unfortunate licence but maybe upstream would be willing to replace those assets?

You could also separate out the assets into a different package to ensure that Hydra can still build the actual engine.

Reply:
Thanks for weighing in. after talking to the author i will wait for the next release, which should resolve the licensing problematic

Reply:
In OutFly [v0.11.0](https://codeberg.org/outfly/outfly/releases/tag/v0.11.0), all non-free assets have been replaced by free ones.

Reply:
thanks for the update and resolving this so quickly!
i will try to finish this after rust 1.81 makes it into nixpkgs master